daily-co/daily-demos
# Demo/react with Stream Chat

# Overview 
While we built out a simple chat demo using `sendAppMessage()`, we have yet to build a demo featuring a third-party chat API integration, something our users have asked for in the past. 

This PR solves for that! Check out README.md for more detail on how it works. 

# Specific feedback I'd appreciate: 
* **CSS**: I just want the height of the .str-chat in `ChatWidget.css` to be a bit shorter. Any pro tips? I used Elements selector to try to find the right Stream Chat element to style, but maybe I'm missing one? Width responded to me, but height and border-radius did not. 
* **Comments**: I added comments where we added and changed things in the code. Please shout if you think I missed anything major to flag. 
* **Glitch server**: I left links to my real server in there for now, with a comment that devs should replace with their own. Does this seem reasonable? Maybe after team review, I could delete real server if we think it's an issue. 
* **props.displayChat**: Is the way I'm doing this in `ChatWidget.js` reasonable? 
* **Bells and whistles**: Is it okay that the chat icon in the tray doesn't currently light up when a message is received? I think this is fine for now, but I can suggest that a reader wire up state to change and watch for this. 
* **user presence**: Thanks for flagging this, Phil! The channel is now watching for that in `ChatWidget.js.` I just had to pass in a param. 

# Steps to review 
1. Make sure you're in the daily-demos repository
2. `git checkout react-demo-stream-chat`
3. `cd static-demos`
4. npm run dev or npm run start
5. In your browser, head to `http://localhost:8080`
6. Start a call. 
7. Copy the link to another tab. 
8. Click the chat icon. 
9. Send a message. 
10. Make sure you see it in the next tab. 

Thanks a mil for feedback! 
![giphy](https://user-images.githubusercontent.com/28582074/91485092-1c25d500-e85f-11ea-96e0-ecd2eb59db2f.gif)
